### PR TITLE
replace deprecated method

### LINKED ISSUE #832: <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
Replace with MatcherAssert.assertThat

[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java:[101,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/GenericArchivesVerifier.java:[47,10] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/GenericArchivesVerifier.java:[57,10] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java:[71,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java:[86,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java:[71,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java:[86,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/AppEngineDeployResultTest.java:[112,6] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/AppEngineDeployResultTest.java:[122,6] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java:[66,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java:[111,12] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/CloudSdkTest.java:[141,4] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/GcloudStructuredLogTest.java:[103,6] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
[WARNING] /usr/local/google/home/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/components/WindowsBundledPythonCopierTest.java:[85,10] [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated

### PR DESCRIPTION
**Pull request by elharo**
@chanseokoh fixes #832

### CODE REVIEW

File: src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java [hunk 1]
```diff
@@ -63,9 +62,7 @@ public void testLogin_withBadUser() {
       new Auth(gcloudRunner).login(testUsername);
       Assert.fail("Should have failed with bad user.");
     } catch (AppEngineException e) {
-      Assert.assertThat(
-          e.getMessage(), CoreMatchers.containsString("Invalid email address: " + testUsername));
-      // pass
+      Assert.assertTrue(e.getMessage().contains("Invalid email address: " + testUsername));
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java]: could this not be converted to `org.hamcrest.MatcherAssert.assertThat` ?

Comment (loosebazooka) [reply src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java]: using http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/StringContains.html

```
import org.hamcrest.core.StringContains;

...
...

assertThat(e.getMessage(), StringContains.containsString("Invalid email address: " + testUsername));
```

**Comment by elharo** [comment]
weird failure: RROR] Failed to execute goal org.apache.felix:maven-bundle-plugin:3.5.0:manifest (bundle-manifest) on project appengine-plugins-core: Execution bundle-manifest of goal org.apache.felix:maven-bundle-plugin:3.5.0:manifest failed: Plugin org.apache.felix:maven-bundle-plugin:3.5.0 or one of its dependencies could not be resolved: Could not transfer artifact biz.aQute.bnd:biz.aQute.bndlib:jar:3.5.0 from/to google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/): Connection reset -> [Help 1]

[ERROR]

File: src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java (hunk 1 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java]: I can do that if you want, but the code's a little less clear and I suspect we'll have to  change it again in a few months anyway. The more we can do with the most fundamental classes the better.

Comment (loosebazooka) [reply src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java]: I would prefer assertThat, it provides a much better error message on failure. And I'm not convinced it's less readable.

Comment (elharo) [reply src/test/java/com/google/cloud/tools/appengine/operations/AuthTest.java]: done

File: src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/GcloudStructuredLogTest.java [hunk 2]
```diff
@@ -99,8 +98,10 @@ public void testParse_inputMalformed() {
     try {
       GcloudStructuredLog.parse("non-JSON");
       fail();
-    } catch (JsonParseException e) {
-      assertThat(e.getMessage(), CoreMatchers.containsString("JsonSyntaxException"));
+    } catch (JsonParseException expected) {
+      String message = expected.getMessage();
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/GcloudStructuredLogTest.java]: Seems like this should remain assertThat (with new package) with containsString, would remove need for null check.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java [hunk 3]
```diff
@@ -68,8 +67,9 @@ public void testZipSlipVulnerability_windows() throws URISyntaxException {
       tarGzExtractorProvider.extract(testArchive, extractionRoot, mockProgressListener);
       Assert.fail("IOException expected");
     } catch (IOException ex) {
-      Assert.assertThat(
-          ex.getMessage(), Matchers.startsWith("Blocked unzipping files outside destination: "));
+      String message = ex.getMessage();
+      Assert.assertNotNull(message);
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java]: Seems like this should remain assertThat (with new package) with containsString, would remove need for null check.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java [hunk 4]
```diff
@@ -83,8 +83,9 @@ public void testZipSlipVulnerability_unix() throws URISyntaxException {
...
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java]: Seems like this should remain assertThat (with new package) with containsString, would remove need for null check.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java [hunk 5]
```diff
@@ -68,8 +67,9 @@ public void testZipSlipVulnerability_windows() throws URISyntaxException {
       zipExtractorProvider.extract(testArchive, extractionRoot, mockProgressListener);
       Assert.fail("IOException expected");
     } catch (IOException ex) {
-      Assert.assertThat(
-          ex.getMessage(), Matchers.startsWith("Blocked unzipping files outside destination: "));
+      String message = ex.getMessage();
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java]: Seems like this should remain assertThat (with new package) with containsString, would remove need for null check.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java [hunk 6]
```diff
@@ -83,8 +83,9 @@ public void testZipSlipVulnerability_unix() throws URISyntaxException {
...
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java]: Seems like this should remain assertThat (with new package) with containsString, would remove need for null check.

File: src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/GcloudStructuredLogTest.java (hunk 2 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/appengine/operations/cloudsdk/serialization/GcloudStructuredLogTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java (hunk 3 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java (hunk 4 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProviderTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java (hunk 5 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java (hunk 6 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/ZipExtractorProviderTest.java]: done

**Pull request merged**